modules/services/dnsmasq.nix

```diff
@@ -0,0 +1,71 @@
+{ pkgs, lib, nglib, config, ... }:
+with lib; with nglib;
+let
+  cfg = config.services.dnsmasq;
+in
+{
+  options.services.dnsmasq = {
+    enable = mkEnableOption "dnsmasq";
+    package = mkPackageOption pkgs "dnsmasq" { };
+
+    user = mkOption {
+      description = "dnsmasq user";
+      type = types.str;
+      default = "dnsmasq";
+    };
+
+    group = mkOption {
+      description = "dnsmasq group";
+      type = types.str;
+      default = "dnsmasq";
+    };
+
+    configuration = mkOption {
+      type = types.str;
```

Comment:
This must be at least `types.lines` (`str` does not support merging), ideally a custom option type with a custom generator, like `nginx`, `apache` and others.